portability/file.cc

```diff
@@ -186,9 +195,64 @@ void toku_set_func_pread (ssize_t (*pread_fun)(int, void *, size_t, off_t)) {
     t_pread = pread_fun;
 }
 
-void
-toku_os_full_write (int fd, const void *buf, size_t len) {
-    const char *bp = (const char *) buf;
+int toku_os_delete_with_source_location(const char *name,
+                                        const char *src_file,
+                                        uint src_line) {
+    int result;
+
+    toku_io_instrumentation io_annotation;
+    toku_instr_file_name_close_begin(io_annotation,
+                                     *tokudb_file_data_key,
+                                     toku_instr_file_op::file_delete,
+                                     name,
+                                     src_file,
+                                     src_line);
+    result = unlink(name);
```

Comment:
As during the last review round, I believe that declaration at initialisation, with "const" and "auto" as appropriate is the right style in C++11 and forward. This is not my decision to make and it's OK to ignore, but please make sure to discuss this with George going forward.